### PR TITLE
SQL: MATCH and QUERY documentation; one list of functions

### LINKED ISSUE #35076: SQL: document the MATCH and QUERY full-text queries
MATCH and QUERY functions are tangentially mentioned in the full-text functions sections when talking about SCORE, but they don't have their own dedicated section.

### LINKED ISSUE #36885: SQL: Make a one-page list of all implemented functions
To ease usage, SQL documentation should have an easy to access page listing all implemented functions so it can be easily searchable.

### PR DESCRIPTION
**Pull request by astefan**
This PR adds documentation for MATCH and QUERY function predicates, updates the Functions & Operators page to include the list of all available functions and polishes (&rearranges) some of the other functions pages.

Fixes https://github.com/elastic/elasticsearch/issues/35076.
Fixes https://github.com/elastic/elasticsearch/issues/36885.

### CODE REVIEW

File: docs/reference/sql/functions/search.asciidoc [hunk 1]
```diff
@@ -8,6 +8,109 @@ when the `MATCH` or `QUERY` predicates are being used.
...
+and <<query-dsl-multi-match-query,multi_match>> {es} queries.
+
+The first parameter is the field or fields to match against. In case it receives one value only, {es-sql} will use a `match` query to perform the search:
+
+["source","sql",subs="attributes,callouts,macros"]
+----
+include-tagged::{sql-specs}/docs/docs.csv-spec[simpleMatch]
+----
+
+However, it can also receive a list of fields and their corresponding optional `boost` value. In this case, {es-sql} will use a
```
Comment (costin) [inline docs/reference/sql/functions/search.asciidoc]: I would add a basic explanation on boosting does in this example for SQL folks that might be familiar with it.

File: docs/reference/sql/functions/search.asciidoc [hunk 2]
```diff
@@ -8,6 +8,109 @@ when the `MATCH` or `QUERY` predicates are being used.
...
+["source","sql",subs="attributes,callouts,macros"]
+----
+include-tagged::{sql-specs}/docs/docs.csv-spec[multiFieldsMatch]
+----
+
+Both options above can be used in combination with the optional third parameter of the `MATCH()` predicate, where one can specify
+additional configuration parameters (separated by semicolon `;`) for either `match` or `multi_match` queries. For example:
+
+["source","sql",subs="attributes,callouts,macros"]
+----
```
Comment (costin) [inline docs/reference/sql/functions/search.asciidoc]: Same here - explain what the example tries to achieve.

File: docs/reference/sql/functions/index.asciidoc [hunk 3]
```diff
@@ -7,14 +7,115 @@
 
 * <<sql-operators, Operators>>
```
Comment (costin) [inline docs/reference/sql/functions/index.asciidoc]: 👍

**Review by costin — APPROVED** [review]
Excellent work!
My only comment would be to add a comment to `LIKE` and `RLIKE` section mentioning the advantages of full-text queries: faster, more powerful and accurate.

**Comment by astefan** [comment]
@costin I've created https://github.com/elastic/elasticsearch/issues/40508 for completing LIKE/RLIKE docs sections.

**Review by matriv — APPROVED** [review]
LGTM. Great work here!

**Pull request merged**